src/ewilgs/save_data.py

```diff
@@ -21,11 +21,23 @@ def add_frame(frame, username=None, application=None) -> None:
     """Adds one json frame to the downlink table"""
     downlink_entry = Downlink()
 
-    # assign values
-    if 'frequency' in frame or frame['frequency'] is not None:
-        downlink_entry.frequency = frame['frequency']
+    # check if the frame exists and it is a HEX string
+    non_hex = re.match("[^0-9A-Fa-f]", frame['frame'])
+    if non_hex:
+        raise ValueError("Invalid frame, not an hexadecimal value.")
+
+    # assign the frame HEX values
     downlink_entry.frame = frame['frame']
-    if 'qos' in frame or frame['qos'] is not None:
+
+    # assign the timestamp
+    downlink_entry.receive_time = parse_datetime(frame['timestamp']).astimezone(pytz.utc)
```

Comment:
Shall we also raise exceptions if the submission is missing the frame and/or timestamp field?

Reply:
It actually happens already in [views.py](https://github.com/iiacoban42/DelfiTLM/blob/92e8e4e27a258597557af1f7bb8aff16d6cd7c05/src/ewilgs/views.py#L46).